crates/pipeline_manager/src/main.rs

```diff
@@ -311,14 +316,29 @@ fn run(config: ManagerConfig) -> AnyResult<()> {
 
         let state = WebData::new(ServerState::new(config, db, compiler).await?);
 
-        let server = HttpServer::new(move || {
-            let app = App::new()
-                .wrap(Logger::default())
-                .wrap(Condition::new(dev_mode, actix_cors::Cors::permissive()));
-
-            build_app(app, state.clone(), openapi.clone())
-        });
-        server.listen(listener)?.run().await?;
+        if use_auth {
+            let server = HttpServer::new(move || {
+                let closure = |req, bearer_auth| {
+                    auth::auth_validator(auth::aws_auth_config(), req, bearer_auth)
+                };
+                let auth_middleware = HttpAuthentication::bearer(closure);
+                let app = App::new()
+                    .wrap(Logger::default())
+                    .wrap(Condition::new(dev_mode, actix_cors::Cors::permissive()))
+                    .wrap(auth_middleware);
```

Comment:
instead of duplicating code maybe just have 

if use_auth {
  app = app.wrap(auth_middleware)
}

as rest of code seems the same between if and else branch?

Reply:
`app.wrap(auth_middleware)` changes the type of `app`, so that doesn't work.